complex.js

```diff
@@ -456,6 +456,18 @@
         return Complex['ONE'];
       }
 
+      if (!z.isFinite()) {
+        return Complex['NAN'];
+      }
+
+      if (!this.isFinite()) {
+        return z['re'] < 0
+          ? Complex['ZERO']
+          : z['im'] === 0 && z['re'] > 0 // should we epsilon ball here
```

Comment:
Should the imaginary part of z be compared exactly to `0` here or should `Complex['EPISLON']` be used?